services/src/main/scala/cromwell/services/healthmonitor/HealthMonitorServiceActor.scala

```diff
@@ -0,0 +1,119 @@
+package cromwell.services.healthmonitor
+
+import java.util.concurrent.TimeoutException
+
+import akka.actor.{Actor, Scheduler, Timers}
+import akka.pattern.{after, pipe}
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.language.postfixOps
+import HealthMonitorServiceActor._
+import com.typesafe.config.Config
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import net.ceedubs.ficus.Ficus._
+
+trait HealthMonitorServiceActor extends Actor with LazyLogging with Timers {
+  val serviceConfig: Config
+  def subsystems: List[MonitoredSubsystem]
+
+  implicit val ec: ExecutionContext = context.system.dispatcher
+
+  lazy val futureTimeout: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-timeout").getOrElse(DefaultFutureTimeout)
+  lazy val staleThreshold: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.status-ttl").getOrElse(DefaultStaleThreshold)
+
+  override def preStart(): Unit = {
+    val sweepTime = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-refresh-time").getOrElse(DefaultSweepTime)
+    timers.startPeriodicTimer(CheckTickKey, CheckAll, sweepTime)
+    logger.info("Starting health monitor with the following checks: " + subsystems.map(_.name).mkString(", "))
+  }
+
+  /**
+    * Contains each subsystem status along with a timestamp of when the entry was made so we know when the status
+    * goes stale. Initialized with unknown status.
+    */
+  private var data: Map[MonitoredSubsystem, CachedSubsystemStatus] = {
```

Comment:
perhaps not the most expressive choice of variable name

Reply:
You should be code reviewing Rawls too :P